src/mappings/twab-controller.ts

```diff
@@ -4,88 +4,60 @@ import {
-
-    const timestamp = event.block.timestamp;
-    const twab = createTwab(vault, user, timestamp);
-    setTwab(twab, account, amount, delegateAmount, isNew, timestamp);
-
-    twab.save();
-    account.save();
+    const { vault, user, amount, delegateAmount } = event.params;
+    const updateId = generateUniqueLogId(event);
+    createAccountBalanceUpdate(updateId, vault, user, amount, delegateAmount, event.block.timestamp);
 }
 
 export function handleDecreasedBalance(event: DecreasedBalance): void {
-    const vault = event.params.vault;
-    const user = event.params.user;
-    const amount = event.params.amount;
-    const delegateAmount = event.params.delegateAmount;
-    const isNew = event.params.isNew;
-
-    const account = loadOrCreateAccount(vault, user);
-    decreaseBalance(account, amount);
-    decreaseDelegateBalance(account, delegateAmount);
-
-    const timestamp = event.block.timestamp;
-    const twab = createTwab(vault, user, timestamp);
-    setTwab(twab, account, amount, delegateAmount, isNew, timestamp);
-
-    twab.save();
-    account.save();
+    const { vault, user, amount, delegateAmount } = event.params;
+    const updateId = generateUniqueLogId(event);
+    createAccountBalanceUpdate(updateId, vault, user, amount.neg(), delegateAmount.neg(), event.block.timestamp);
 }
 
 export function handleIncreasedTotalSupply(event: IncreasedTotalSupply): void {
-    const vault = event.params.vault;
-    const amount = event.params.amount;
-
-    const vaultEntity = loadOrCreateVault(vault);
-    increaseTotalSupply(vaultEntity, amount);
-
-    vaultEntity.save();
+    const { vault, amount, delegateAmount } = event.params;
+    const updateId = generateUniqueLogId(event);
+    createVaultBalanceUpdate(updateId, vault, amount, delegateAmount, event.block.timestamp);
 }
 
 export function handleDecreasedTotalSupply(event: DecreasedTotalSupply): void {
-    const vault = event.params.vault;
-    const amount = event.params.amount;
-
-    const vaultEntity = loadOrCreateVault(vault);
-    decreaseTotalSupply(vaultEntity, amount);
-
-    vaultEntity.save();
+    const { vault, amount, delegateAmount } = event.params;
+    const updateId = generateUniqueLogId(event);
+    createVaultBalanceUpdate(updateId, vault, amount.neg(), delegateAmount.neg(), event.block.timestamp);
 }
 
 export function handleDelegated(event: Delegated): void {
-    const vault = event.params.vault;
-    const delegate = event.params.delegator;
-    const delegatee = event.params.delegate;
+    const { vault, delegator, delegate } = event.params;
 
+    const delegatorAccount = loadOrCreateAccount(vault, delegator);
     const delegateAccount = loadOrCreateAccount(vault, delegate);
-    const delegateeAccount = loadOrCreateAccount(vault, delegatee);
 
-    setDelegatee(delegateAccount, delegateeAccount.id);
+    delegatorAccount.delegate = delegateAccount.id;
 
+    delegatorAccount.save();
     delegateAccount.save();
```

Comment:
We're not making any changes to the `delegateAccount` so there is no need to `.save()` here

Reply:
Removed with: 5e9471a